Ginger/GingerCoreNET/Application Models/Delta/APIDelta/APIDeltaUtils.cs

```diff
@@ -0,0 +1,125 @@
+
+            //for (int i = 0; i < learnedAPIModelsList.Count; i++)
+            foreach (ApplicationAPIModel apiModelLearned in learnedAPIModelsList)
+            {
+                //ApplicationAPIModel apiModelLearned = learnedAPIModelsList[i];
+
+                DeltaAPIModel apiModelDelta = new DeltaAPIModel();
+                apiModelDelta.learnedAPI = apiModelLearned;
+                List<ApplicationAPIModel> matchingAPIModels;
+                if (existingAPIModels == null)
+                {
+                    if (apiModelLearned.APIType == ApplicationAPIUtils.eWebApiType.SOAP)
+                        matchingAPIModels = existingAPIModelsList.Where(m => m.EndpointURL.Equals(apiModelLearned.EndpointURL, StringComparison.OrdinalIgnoreCase) && m.APIType == apiModelLearned.APIType && m.SOAPAction.Equals(apiModelLearned.SOAPAction)).ToList();
+                    else
+                        matchingAPIModels = existingAPIModelsList.Where(m => m.EndpointURL.Equals(apiModelLearned.EndpointURL, StringComparison.OrdinalIgnoreCase) && m.APIType == apiModelLearned.APIType && m.RequestType == apiModelLearned.RequestType).ToList();
+                }
+                else
+                {
+                    matchingAPIModels = existingAPIModels.ToList();
+                }
+
+                if (matchingAPIModels != null && matchingAPIModels.Count > 0)
+                {
+                    ApplicationAPIModel matchingModel = matchingAPIModels[0];
+                    matchingAPIModels = CompareAPIModels(apiModelLearned, matchingAPIModels);
+                    if (matchingAPIModels.Count > 0)
+                    {
+                        apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.Unchanged;         // UNCHANGED
+                        apiModelDelta.matchingAPIModel = matchingAPIModels[0];
+                        apiModelDelta.IsSelected = false;
+                    }
+                    else
+                    {
+                        apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.Modified;          // MODIFIED
+                        apiModelDelta.matchingAPIModel = matchingModel;
+                        apiModelDelta.IsSelected = true;
+                    }
+                }
+                else
+                {
+                    apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.New;                   // NEW API
+                    apiModelDelta.IsSelected = true;
+                }
+
+                deltaAPIModelsList.Add(apiModelDelta);
+            }
+
+            deltaAPIModelsList = new ObservableList<DeltaAPIModel>(deltaAPIModelsList.OrderBy(d => d.comparisonStatus));
+
+            return deltaAPIModelsList;
+        }
+
+        public static ApplicationAPIModel CreateAPIModelObject(ApplicationAPIModel sourceAPIModel)
```

Comment:
this is only for creating delta right?
so need to call it "CreateDeltaAPIModelObject"

Reply:
No. This is used to create copy of API Model. I've moved to APILearnUtils now.